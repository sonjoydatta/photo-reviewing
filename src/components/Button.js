import styled from 'styled-components';
import { hexToRGB } from '@/utils/helpers';
import { COLORS, RADIUS } from './constants';

const Button = styled.button.attrs((props) => ({
  type: props.type || 'button',
  variant: props.variant || 'primary',
  block: props.block || false,
  rounded: props.rounded || false,
}))`
  width: ${({ block }) => (block ? '100%' : 'auto')};
  padding: 0.5rem 1.5rem;
  border: none;
  border-radius: ${({ rounded }) => (rounded ? RADIUS.rounded : 0)};
  background-color: ${({ variant }) => (variant === 'primary' ? COLORS.primary : COLORS.secondary)};
  color: ${COLORS.white};
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.5;
  text-align: center;
  text-decoration: none;
  transition: all 0.4s ease-in-out;
  cursor: pointer;

  &:hover:not(:disabled) {
    background-color: ${({ variant }) => hexToRGB(variant === 'primary' ? COLORS.primary : COLORS.secondary, 0.75)};
  }

  &:disabled {
    cursor: default;
    opacity: 0.5;
  }
`;

export default Button;
