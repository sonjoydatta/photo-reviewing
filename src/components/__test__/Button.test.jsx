import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button>Hello World</Button>);
    expect(tree).toMatchSnapshot();
  });

  it('render with primary variant', () => {
    const tree = renderer.create(<Button variant="primary">Hello World</Button>);
    expect(tree).toMatchSnapshot();
  });

  it('render with secondary variant', () => {
    const tree = renderer.create(<Button variant="secondary">Hello World</Button>);
    expect(tree).toMatchSnapshot();
  });

  it('render with radius variant', () => {
    const tree = renderer.create(<Button rounded>Hello World</Button>);
    expect(tree).toMatchSnapshot();
  });
});
