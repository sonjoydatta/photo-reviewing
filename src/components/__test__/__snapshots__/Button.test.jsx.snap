// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render with primary variant 1`] = `
<button
  className="sc-bczRLJ ypFDz"
  type="button"
>
  Hello World
</button>
`;

exports[`Button component render with radius variant 1`] = `
<button
  className="sc-bczRLJ kNlsHU"
  type="button"
>
  Hello World
</button>
`;

exports[`Button component render with secondary variant 1`] = `
<button
  className="sc-bczRLJ jXFloR"
  type="button"
>
  Hello World
</button>
`;

exports[`Button component renders correctly 1`] = `
<button
  className="sc-bczRLJ ypFDz"
  type="button"
>
  Hello World
</button>
`;
